sklearn/utils/estimator_checks.py

```diff
@@ -1106,60 +1104,105 @@ def check_sample_weights_invariance(name, estimator_orig, kind="ones"):
+    sw[len(y) :] = 0
+    # repeat samples according to weights
+    X_repeated = X_weigthed.repeat(repeats=sw, axis=0)
+    y_repeated = y_weighted.repeat(repeats=sw)
+
+    X_weigthed, y_weighted, sw = shuffle(X_weigthed, y_weighted, sw, random_state=0)
 
     # when the estimator has an internal CV scheme
     # we only use weights / repetitions in a specific CV group (here group=0)
     if "cv" in estimator_orig.get_params():
-        groups2 = np.hstack(
-            [np.full_like(y2, 0), np.full_like(y1, 1), np.full_like(y1, 2)]
+        groups_weighted = np.hstack(
+            [np.full_like(y_weighted, 0), np.full_like(y, 1), np.full_like(y, 2)]
+        )
+        sw = np.hstack([sw, np.ones_like(y), np.ones_like(y)])
+        X_weigthed = np.vstack([X_weigthed, X, X])
+        y_weighted = np.hstack([y_weighted, y, y])
+        splits_weighted = list(
+            LeaveOneGroupOut().split(X_weigthed, groups=groups_weighted)
         )
-        sw2 = np.hstack([sw2, np.ones_like(y1), np.ones_like(y1)])
-        X2 = np.vstack([X2, X1, X1])
-        y2 = np.hstack([y2, y1, y1])
-        splits2 = list(LeaveOneGroupOut().split(X2, groups=groups2))
-        estimator2.set_params(cv=splits2)
-
-        groups1 = np.hstack(
-            [np.full_like(y1, 0), np.full_like(y1, 1), np.full_like(y1, 2)]
+        estimator_weighted.set_params(cv=splits_weighted)
+
+        groups_repeated = np.hstack(
+            [np.full_like(y_repeated, 0), np.full_like(y, 1), np.full_like(y, 2)]
+        )
+        X_repeated = np.vstack([X_repeated, X, X])
+        y_repeated = np.hstack([y_repeated, y, y])
+        splits_repeated = list(
+            LeaveOneGroupOut().split(X_repeated, groups=groups_repeated)
         )
-        X1 = np.vstack([X1, X1, X1])
-        y1 = np.hstack([y1, y1, y1])
-        splits1 = list(LeaveOneGroupOut().split(X1, groups=groups1))
-        estimator1.set_params(cv=splits1)
+        estimator_repeated.set_params(cv=splits_repeated)
 
-    y1 = _enforce_estimator_tags_y(estimator1, y1)
-    y2 = _enforce_estimator_tags_y(estimator2, y2)
+    y_weighted = _enforce_estimator_tags_y(estimator_weighted, y_weighted)
+    y_repeated = _enforce_estimator_tags_y(estimator_repeated, y_repeated)
 
-    estimator1.fit(X1, y=y1, sample_weight=None)
-    estimator2.fit(X2, y=y2, sample_weight=sw2)
+    estimator_repeated.fit(X_repeated, y=y_repeated, sample_weight=None)
+    estimator_weighted.fit(X_weigthed, y=y_weighted, sample_weight=sw)
 
     for method in ["predict", "predict_proba", "decision_function", "transform"]:
         if hasattr(estimator_orig, method):
-            X_pred1 = getattr(estimator1, method)(X1)
-            X_pred2 = getattr(estimator2, method)(X1)
+            X_pred1 = getattr(estimator_repeated, method)(X)
+            X_pred2 = getattr(estimator_weighted, method)(X)
+            err_msg = (
+                f"For {name}.{method} sample_weight is not equivalent "
```

Comment:
I think that specifying the `method` is misleading. Usually the discrepancy comes from fit already. The coef_ attribute may be different for instance, which results in different predictions, but it doesn't mean that the bug is in predict.

Reply:
Still it might be interesting to include the method name in the message to indicate how the discrepancy was detected without implying that the source of discrepancy comes from that prediction method.

```python
    f"Comparing the output of {name}.{method} revealed that fitting with `sample_weight`"
    "is not equivalent to to fitting with removed or repeated samples."
```